src/js/components/TabForm.js

```diff
@@ -0,0 +1,201 @@
+import classNames from 'classnames';
+import {Form, Tooltip} from 'reactjs-components';
+import GeminiScrollbar from 'react-gemini-scrollbar';
+import React from 'react';
+
+import GeminiUtil from '../utils/GeminiUtil';
+import SideTabs from './SideTabs';
+
+const METHODS_TO_BIND = [
+  'getTriggerSubmit', 'validateForm', 'handleTabClick', 'handleExternalSubmit'
+];
+
+class TabForm extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {currentTab: '', renderGemini: false};
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+
+    this.triggerSubmit = function () {};
+    this.isValidated = true;
+  }
+
+  componentWillMount() {
+    this.model = {};
+    this.submitMap = {};
+
+    this.setState({
+      currentTab: Object.keys(this.props.definition)[0]
+    });
+
+    this.props.getTriggerSubmit(this.handleExternalSubmit);
+  }
+
+  componentDidMount() {
+    this.setState({renderGemini: true});
+  }
+
+  componentDidUpdate() {
+    // Timeout necessary due to modal content height updates on did mount
+    setTimeout(() => {
```

Comment:
You could consider the MutationObserver to listen on render changes with a save way.
https://developer.mozilla.org/de/docs/Web/API/MutationObserver

Just a suggestion, feel free to ignore or implement in another PR.